asyncgit/src/sync/cred.rs

```diff
@@ -70,6 +80,69 @@ pub fn extract_username_password(
 	})
 }
 
+fn use_credential_store(config: &Config) -> bool {
+	config
+		.get_entry("credential.helper")
+		.ok()
+		.as_ref()
+		.and_then(git2::ConfigEntry::value)
+		.map(|val| val == "store")
+		.unwrap_or_default()
+}
+
+// tries calling:
+// printf "protocol=https\nhost=github.com\n" | git credential fill
+// see https://git-scm.com/book/en/v2/Git-Tools-Credential-Storage
+// TODO: use input stream
+fn git_credential_fill(url: &str) -> Option<BasicAuthCredential> {
+	scope_time!("git_credential_fill");
+
+	let url = url::Url::parse(url).ok()?;
+
+	let host = url.domain()?;
+	let protocol = url.scheme();
+
+	let cmd = format!("protocol={}\nhost={}\n", protocol, host);
+	let cmd =
+		format!("printf \"{}\" | git credential-store get", cmd);
```

Comment:
I would change to this

```suggestion
		format!("printf \"{}\" | git credential fill", cmd);
```

Reply:
problem with that is that it breaks if you don't have credentials stored yet. not sure why though (yet)

Reply:
You are right - I had not checked that. 
`git credential fill` asks for the credentials interactively...

Hmm- I do not see any other solution yet.